lib/internal/file.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+const {
+  DateNow,
+  NumberIsNaN,
+  ObjectDefineProperties,
+  ObjectPrototypeHasOwnProperty,
+  Symbol,
+  SymbolToStringTag,
+} = primordials;
+
+const {
+  Blob,
+} = require('internal/blob');
+
+const {
+  customInspectSymbol: kInspect,
+  emitExperimentalWarning,
+  kEnumerableProperty,
+  kEmptyObject,
+  toUSVString,
+} = require('internal/util');
+
+const {
+  codes: {
+    ERR_INVALID_THIS,
+    ERR_MISSING_ARGS,
+  },
+} = require('internal/errors');
+
+const {
+  inspect,
+} = require('internal/util/inspect');
+
+const kState = Symbol('kState');
+
+function isFile(object) {
+  return object?.[kState] !== undefined || object instanceof Blob;
+}
+
+class File extends Blob {
+  constructor(fileBits, fileName, options = kEmptyObject) {
+    emitExperimentalWarning('buffer.File');
+
+    if (arguments.length < 2) {
+      throw new ERR_MISSING_ARGS('fileBits', 'fileName');
+    }
+
+    super(fileBits, options);
```

Comment:
I believe before this line we should have a `validateString(fileName, 'filename')` check. `toUSVString` is too late to check.

Reply:
fileName doesn't necessarily have to be a string - just any value that can be stringified and doesn't contain any unicode scalar values

Reply:
I see. Thanks for the explanation